taqueria-vscode-extension/src/lib/gui/ArtifactsDataProvider.ts

```diff
@@ -0,0 +1,50 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { HasFileName, HasRefresh, VsCodeHelper } from '../helpers';
+
+export class ArtifactsDataProvider implements vscode.TreeDataProvider<ArtifactTreeItem>, HasRefresh {
+	constructor(
+		private workspaceRoot: string,
+		private helper: VsCodeHelper,
+	) {}
+
+	getTreeItem(element: ArtifactTreeItem): vscode.TreeItem {
+		return element;
+	}
+
+	async getChildren(element?: ArtifactTreeItem): Promise<ArtifactTreeItem[]> {
+		const artifacts = await vscode.workspace.findFiles('artifacts/*.*', '**/node_modules/**');
```

Comment:
This will work well for Ligo and Archetype artifacts (that are single `.tz` files).
Smartpy creates folders with multiple files in it. This will mess up our artifacts view.

I don't know how we should handle this. I guess we should take the folder's name? @russellmorton

Reply:
I think SmartPY is a separate milestone. We should track this issue in another ticket to make sure we won't forget about it 😉